tests/test_storage_full_path.py

```diff
@@ -0,0 +1,79 @@
+# Copyright (c) 2020 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+import os
+import yaml
+import re
+
+from tests.lib import common
+from tests.config import settings
+
+if settings.DISTRO == 'SLES_CaaSP':
+    from tests.lib.rook.ses import RookSes as RookCluster
+else:
+    from tests.lib.rook.upstream import RookCluster as RookCluster  # type: ignore  # noqa: E501
+
+
+logger = logging.getLogger(__name__)
+
+
+def set_storage_pattern(rook_cluster, pattern):
+
+    # create a working-copy of cluster.yaml
+    cluster_working = os.path.join(rook_cluster.ceph_dir, 'cluster_test.yaml')
+    cluster_original = os.path.join(rook_cluster.ceph_dir, 'cluster.yaml')
+
+    cluster_orig_f = open(cluster_original, 'r')
+    cluster_object = yaml.full_load(cluster_orig_f)
+
+    cluster_object['spec']['storage']['devices'] = [{'name': pattern}]
+
+    cluster_work_f = open(cluster_working, 'a')
+    cluster_work_f.write(yaml.dump(cluster_object, Dumper=yaml.Dumper))
+
+    cluster_orig_f.close()
+    cluster_work_f.close()
+
+    return cluster_working
+
+
+def test_storage_full_path_pattern(workspace, hardware, kubernetes):
+    # pre-install rook
+    with RookCluster(workspace, kubernetes) as rook_cluster:
+        rook_cluster.build()
+        rook_cluster.preinstall()
+        rook_cluster.install_rook()
```

Comment:
When exiting this context the rookcluster could be destroyed. I think you want the whole test to be inside this context.

Reply:
Not necessary anymore at all. Removed it completely.